tests/Feature/Admin/IncorrectTagsTest.php

```diff
@@ -0,0 +1,224 @@
+use Tests\TestCase;
+
+class IncorrectTagsTest extends TestCase
+{
+    use HasPhotoUploads;
+
+    /** @var User */
+    protected $admin;
+    /** @var User */
+    protected $user;
+    /** @var Photo */
+    protected $photo;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        Storage::fake('s3');
+        Storage::fake('bbox');
+
+        $this->setImagePath();
+
+        /** @var User $admin */
+        $this->admin = User::factory()->create(['verification_required' => false]);
+
+        $this->admin->assignRole(Role::create(['name' => 'admin']));
+
+        $this->user = User::factory()->create(['verification_required' => true]);
+
+        // User uploads an image -------------------
+        $this->actingAs($this->user);
+
+        $imageAndAttributes = $this->getImageAndAttributes();
+
+        $this->post('/submit', ['file' => $imageAndAttributes['file']]);
+
+        $this->photo = $this->user->fresh()->photos->last();
+    }
+
+    public function test_an_admin_can_mark_photos_as_incorrectly_tagged()
+    {
+        // User tags the image
+        $this->actingAs($this->user);
+
+        $this->post('/add-tags', [
+            'photo_id' => $this->photo->id,
+            'presence' => true,
+            'tags' => [
+                'smoking' => [
+                    'butts' => 3
+                ]
+            ]
+        ]);
+
+        $this->photo->refresh();
+
+        $smokingId = $this->photo->smoking_id;
+
+        // We make sure xp and tags are correct
+        $this->assertEquals(4, $this->user->xp);
+        $this->assertInstanceOf(Smoking::class, $this->photo->smoking);
+
+        // Admin marks the tagging as incorrect -------------------
+        $this->actingAs($this->admin);
+
+        $this->post('/admin/incorrect', ['photoId' => $this->photo->id])
+            ->assertOk();
+
+        $this->user->refresh();
+        $this->photo->refresh();
+
+        // Assert xp is decreased, and tags are cleared
+        $this->assertEquals(1, $this->user->xp);
+        $this->assertEquals(0, $this->user->count_correctly_verified);
+        $this->assertEquals(0, $this->photo->verification);
+        $this->assertEquals(0, $this->photo->verified);
+        $this->assertEquals(0, $this->photo->total_litter);
+        $this->assertNull($this->photo->result_string);
+        $this->assertNull($this->photo->smoking_id);
+        $this->assertDatabaseMissing('smoking', ['id' => $smokingId]);
+    }
+
+    public function test_leaderboards_are_updated_when_an_admin_marks_tagging_incorrect_from_a_user_with_public_name()
+    {
+        $country = Country::find($this->photo->country_id)->country;
+        $state = State::find($this->photo->state_id)->state;
+        $city = City::find($this->photo->city_id)->city;
+
+        Redis::del("{$country}:Leaderboard");
+        Redis::del("{$country}:{$state}:Leaderboard");
+        Redis::del("{$country}:{$state}:{$city}:Leaderboard");
```

Comment:
Will this use the same redis db?

Reply:
I'm not sure about this, I thought the Redis facade automatically faked when on testing. I've run the tests on the staging server, everything worked, but didn't check for Redis side effects. Let me take a look.

Reply:
Yep, it uses the local redis server. To make it use a fake/mock redis instance would require some more work, we can skip it for now. The staging server has a separate redis instance, right?

These tests are not meant to be run on production.

Reply:
Redis production is separate to staging. They live on different servers (aws + DO). Just wanted to check and flag this